### PR TITLE
chore: Add i18n strings for date picker at month granularity

### PR DESCRIPTION
**Pull request by jperals**
### Description

Add i18n files necessary for #1942.

Related API doc: `GaXpAdXYZDe2`

i18n process doc: `U6XMA8nIG5U2`

### How has this been tested?

No specific tests for the new i18n strings. Tests for the actual ARIA labels will be provided as part of #1942.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/i18n/messages/all.de.json
Comment (connorlanigan) [inline src/i18n/messages/all.de.json:58]: This is an incorrect translation ("Costs of the current month"). I'd suggest "Aktueller Monat" instead.

**Review by connorlanigan — CHANGES_REQUESTED** [review]

File: src/i18n/messages/all.de.json
Comment (jperals) [reply src/i18n/messages/all.de.json:58]: Yes, I noticed this and opened a ticket (`D119465504`). Since these strings are not used until #1942, I think this could be still merged in order to unblock publishing that PR, but otherwise I'll do that anyway with the changes in this PR once it's ready with a couple more refactors. One way or the other, the translation fix will have to be integrated before merging #1942.

**Comment by jperals** [comment]
Closing in favor of https://github.com/cloudscape-design/components/pull/1942

**Pull request closed without merging**
